src/aiai_eval/evaluator.py

```diff
@@ -106,61 +107,96 @@ def evaluate(
 
         Returns:
             dict:
-                A nested dictionary of the benchmark results. The keys are the names of
+                A nested dictionary of the evaluation results. The keys are the names of
                 the datasets, with values being new dictionaries having the model IDs
                 as keys.
         """
-        pass
+        # Prepare the model IDs
+        model_ids = self._prepare_model_ids(model_id)
+
+        # Evaluate all the models in `model_ids` on all the datasets in `dataset_tasks`
+        dataset_tasks = self.evaluation_config.dataset_tasks
+        for dataset_task in dataset_tasks:
+            for m_id in model_ids:
+                self._evaluate_single(
+                    dataset_task=dataset_task,
+                    model_id=m_id,
+                )
+
+        # Save the evaluation results
+        if self.evaluation_config.save_results:
+            output_path = Path.cwd() / "aiai_eval_results.json"
+            with output_path.open("w") as f:
+                json.dump(self.evaluation_results, f)
+
+        return self.evaluation_results
 
     def _prepare_model_ids(
         self,
         model_id: Optional[Union[Sequence[str], str]],
     ) -> Sequence[str]:
-        """Prepare the model ID(s) to be benchmarked.
-
+        """Prepare the model ID(s) to be evaluated.
         Args:
-            model_id (str or list of str):
-                The model ID(s) of the models to be evaluated.
-
+            model_id (str, list of str or None):
+                The model ID(s) of the models to evaluate. If None then all model IDs
+                will be retrieved.
```

Comment:
Do we really want to benchmark all models? What does that even mean? In ScandEval this makes sense as we can uniformly benchmark all pretrained models, but if we want to benchmark all hatespeech detection models then there's no clear way to get these from the HF Hub.

Reply:
Agreed, it is not clear how to interpret all models.